src/parser/Literal.h

```diff
@@ -49,7 +49,38 @@ class Literal {
   // character. Throws an exception if the literal has no language tag.
   NormalizedStringView getLanguageTag() const;
 
-  // Return the datatype of the literal, if available, withour leading ^^
+  // Return the datatype of the literal, if available, without leading ^^
   // prefix. Throws an exception if the literal has no datatype.
   Iri getDatatype() const;
+
+  // Create a new Literal without datatype or language tag
+  //   Content is expected to be surrounded by quotation marks.
+  static Literal literalWithQuotes(const std::string& contentWithQuotes);
+
+  // Create a new Literal without datatype or language tag
+  //   Content is expected to not be surrounded by quotation marks.
+  static Literal literalWithoutQuotes(const std::string& contentWithoutQuotes);
+
+  // Create a new Literal with datatype
+  //   Content is expected to be surrounded by quotation marks.
+  static Literal literalWithQuotesWithDatatype(
+      const std::string& contentWithQuotes, Iri datatype);
+
+  // Create a new Literal with datatype
+  //   Content is expected to not be surrounded by quotation marks.
+  static Literal literalWithoutQuotesWithDatatype(
+      const std::string& contentWithoutQuotes, Iri datatype);
+
+  // Create a new Literal with language tag
+  //   Content is expected to be surrounded by quotation marks.
+  //   Language Tag can optionally start with an @ character.
+  static Literal literalWithQuotesWithLanguageTag(
+      const std::string& contentWithQuotes, const std::string& languageTag);
+
+  // Create a new Literal with language tag
+  //   Content is expected to not be surrounded by quotation marks.
+  //   Language Tag can optionally start with an @ character.
+  static Literal literalWithoutQuotesWithLanguageTag(
+      const std::string& contentWithoutQuotes, const std::string& languageTag);
```

Comment:
I think many of these functions can be unified:

1. if the second argument becomes `std::optional<Iri, string>` with a default value of `std::nullopt` or something similar, then you have encoded `nothing, datatype, language tag` into a single function, and you only have two functions left (no quotes and quotes).
2. I would call the first argument `rdfContentWith[out]Quotes` and clearly document that they have to be valid literals according to the RDF SPARQL or Turtle grammar, because they are processed according to the escaping rules of these standards.